Feature: domain & rmw options

Juancams opened the pull request:
Hiiii

I've added two new options that I find quite useful when using your plugin, since they make changing the domain ID and rmw_implementation much easier (learning the export lines is difficult, jeje).

##

**ROS Domain ID**
```zsh
~$ rosws domain
Write the ROS DOMAIN ID beetween 0 and 250
~$ rosws domain 5
~$ echo $ROS_DOMAIN_ID 
5
~$ rosws domain 251   
 * ROS DOMAIN ID must be a number between 0 and 250
```

**RMW Implementation**
```zsh
~$ rosws dds
ROS 2 DDS Vendors
cyclonedds  fastdds     zenoh
~$ rosws dds cyclonedds 
~$ echo $RMW_IMPLEMENTATION 
rmw_cyclonedds_cpp
~$ rosws dds potatodds
 * Wrong ROS DDS vendor, must be one of: fastdds, cyclonedds, zenoh
 ```
##

Both have auto-completion and a brief description of what they do. I have also modified the README.md with a small explanation and example of use.

I hope you like the new options added and of course I am open to modifying any changes you suggest.

Regards ^^

Juancams committed:
Added domain option

Signed-off-by: Juancams <jc.manzanares.serrano@gmail.com>

Juancams committed:
Added dds option & some minor changes

Signed-off-by: Juancams <jc.manzanares.serrano@gmail.com>

Butakus commented:
Thanks for the contribution, some extra utility is always nice.

I have a question or change request regarding the DDS setup. Why only fastdds, cyclonedds and zenoh? Why not connextdds or gurummdds? Also, the identifier for fast is "fastdds", but then the RMW name is "rmw_fastrtps", which may be confusing.
I propose allowing any name and just setting the environment variable with the given name. The drawback is that the user must set the full name (e.g. "rmw_cyclonedds"), but the good thing is that it supports any RMW, including implementations not shipped by ROS (like rmw_iceoryx2, for example).

Butakus commented on _rosws:
```diff
@@ -50,6 +56,8 @@ function _rosws() {
     'activate:Set the given workspace as the active one'
     'distro:Set and source the given ROS 2 distro (humble, iron, rolling, etc.)'
+    'domain:Set and source the given ROS DOMAIN ID'
+    'dds:Set and source the given ROS 2 DDS vendor (fastdds, cyclonedds, zenoh'
```
Missing closing ')'

Juancams replied:
fixed in [d5f566d](https://github.com/Butakus/ros2-env/pull/6/commits/d5f566d79c80db015607fc93c1bc63138e240d12)

Juancams committed:
Small changes & fixes

Signed-off-by: Juancams <jc.manzanares.serrano@gmail.com>

Juancams commented:
- I have added that now, for example, instead of exiting the fastdds option, rmw_fastrtps_cpp appears.
- You can add any other dds name, the ones that appear are just suggestions.
- I've also updated the README.md to comment on some of the options and to add any others.
- Fixed the bug you mentioned.

**Example**
```zsh
~$ rosws dds rmw_
Suggestions ROS 2 DDS Vendors
rmw_cyclonedds_cpp  rmw_fastrtps_cpp    rmw_zenoh_cpp
~$ rosws dds potatodds
~$ echo $RMW_IMPLEMENTATION 
potatodds
```

Butakus commented:
LGTM!
I would just change the "dds" naming to "rmw", because now not all middleware implementations are based on DDS.

The implementation is perfect, though.

Juancams committed:
dds -> rmw

Signed-off-by: Juancams <jc.manzanares.serrano@gmail.com>

Juancams commented:
Done!!
